src/backend/distributed/metadata/node_metadata.c

```diff
@@ -1537,6 +1539,25 @@ citus_nodeport_for_nodeid(PG_FUNCTION_ARGS)
 }
 
 
+/*
+ * citus_coordinator_node_id returns the node id of the coordinator node
+ */
+Datum
+citus_coordinator_node_id(PG_FUNCTION_ARGS)
+{
+	CheckCitusVersion(ERROR);
+
+	int coordinatorNodeId = FindCoordinatorNodeId();
+
+	if (coordinatorNodeId == -1)
+	{
+		PG_RETURN_NULL();
```

Comment:
On second thought, maybe we should return 0 here.

Looking at `select * from citus_stat_activity;` on a cluster without the coordinator in the metadata, the coordinator gets (fake) node ID 0.

That way, if the user runs `select * from citus_stat_activity where nodeid = citus_coordinator_node_id()` from the coordinator, it works regardless of whether the coordinator is in the metadata.